bankroll/model/activity.py

```diff
@@ -78,11 +79,30 @@ def price(self) -> Cash:
     def proceeds(self) -> Cash:
         return self.amount - self.fees
 
-    def __str__(self) -> str:
+    @property
+    def action(self) -> str:
         if self.quantity > 0:
-            action = "Buy   "
+            action = "Buy"
         elif TradeFlags.LIQUIDATED in self.flags:
             action = "Sell-L"
         else:
-            action = "Sell  "
-        return f"{self.date.date()} {action} {abs(self.quantity):>9} {self.instrument:21} {self.amount.paddedString(padding=10)} (before {self.fees.paddedString(padding=5)} in fees)"
+            action = "Sell"
+
+        return action
+
+    @staticmethod
+    def dataframeColumns() -> List[str]:
+        return ["Date", "Action", "Quantity", "Instrument", "Amount", "Fees"]
+
+    def dataframeValues(self) -> List[Any]:
+        return [
+            self.date.date(),
+            self.action,
+            abs(self.quantity),
+            self.instrument,
+            self.amount,
+            self.fees,
+        ]
```

Comment:
What you're doing here is basically the same as `dataclasses.asdict`. The issue I think you're going to run into (with that function and this approach) is that the nested models (like `instrument`) should ideally be unpacked before `DataFrame` conversion, or else we've stuck a complex Python object into a `DataFrame`, which will make it really challenging to manipulate.

Reply:
Currently I am ensuring that the `instrument` uses its string representation. This seems sufficient at the moment to get identical output to the activity command for the csv output. Can we revisit this once we need a nested representation?